packages/react-component-library/src/components/DatePicker/utils/isDateDisabled.ts

```diff
@@ -0,0 +1,16 @@
+import { isAfter, isBefore, isEqual } from 'date-fns'
+
+export function isDateDisabled(date, disabledDays) {
```

Comment:
Check out this alternative implementation that is arguably slightly more readable:

```typescript
export const isDateDisabled = (date: Date, disabledDays: Date[]) => {
  return disabledDays.some(disabledDay => {
    const isExactMatch = isEqual(date, disabledDay)
    const isAfterDate = disabledDay.after && isAfter(date, disabledDay.after)
    const isBeforeDate = disabledDay.before && isBefore(date, disabledDay.before)
    
    return isExactMatch || isAfterDate || isBeforeDate
  })
}
```